README.md

```diff
@@ -48,13 +47,49 @@ jobs:
 
 Pull requests created by actions using the default `GITHUB_TOKEN` cannot trigger other workflows.
 
-You can pass an `on_changes_command` which can run tests, checks etc that would usually run when
-a PR is opened.
+### Workarounds
+
+1. Use an `on_changes_command`
+
+This can run tests, checks etc that would usually run when a PR is opened, to ensure that
+the PR is only opened if the checks pass. The PR itself will still not trigger the checks.
+
+2. Use an alternative token
 
 Alternatively, you can pass a token that is allowed to create further workflows and pass it as the
 `token` input to this action.
 
-Or see the `create-pull-request` [docs][2] for other options.
+For example, to create and use a GitHub App token:
+
+```
+  update-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: "opensafely-core/setup-action@v1"
+      with:
+        python-version: "3.11"
+        install-just: true 
+    
+    - uses: actions/create-github-app-token@v1
+      id: generate-token
+      with:
+        app-id: ${{ secrets.CREATE_PR_APP_ID }}
+        private-key: ${{ secrets.CREATE_PR_APP_PRIVATE_KEY }}
```

Comment:
The advantage of using a GH app to create a dynamic token is that the private key doesn't expire, and we are not tied to a specific user's PAT? And maybe that it works across orgs?

If correct, might be worth capturing the motivations in this doc?

Also, do we have a relevant GH app already created? If so, perhaps this example can explicitly include that app id and secret name? And maybe links to the app config page (obvs would need access to use).

Reply:
> The advantage of using a GH app to create a dynamic token is that the private key doesn't expire, and we are not tied to a specific user's PAT? 

Yes. If it's a specific user's PAT, it will also make the PR as that user. When we tried with Peter's PAT, it made all the PRs with him as the author, which also means he can't approve them.

> And maybe that it works across orgs?

Not right now, because I made the github app installable only in the opensafely-core organisation.  Discussed with @inglesp  that we'd try it out like this first.  I think it's safe enough to allow it to be installable in any organisation, because you can't do anything with it without the app id and private key, but those are the 2 options - opensafely-core or any org. We can't limit it to just our 3(4) orgs.  Possibly a reason to keep the app id as a secret as well as the private key?

> Also, do we have a relevant GH app already created? 
Yes. It's in opensafely-core and it's (currently) called "opensafely-core Create PR" (which I'd assumed would change if we decide to make it installable elsewhere as well.